craterpy/classes.py

```diff
@@ -469,3 +475,186 @@ def plot(self, name="", ax=None, alpha=0.2, **kwargs):
+        elif 'planet' in gdf.columns and gdf['planet'].nunique() == 1:
+            body = gdf['planet'].iloc[0]
+            
+        if 'units' in gdf.columns and gdf['units'].nunique() == 1:
+            units = gdf['units'].iloc[0]
+        
+        # Create a working copy to modify
+        data = gdf.copy()
+        
+        # Try to identify latitude column
+        lat_col = None
+        lat_candidates = ['lat', 'latitude', 'latitude_deg', 'lat_deg', 'y']
+        for col in lat_candidates:
+            if col in data.columns:
+                lat_col = col
+                break
+        
+        # Try to identify longitude column
+        lon_col = None
+        lon_candidates = ['lon', 'longitude', 'longitude_deg', 'lon_deg', 'x']
+        for col in lon_candidates:
+            if col in data.columns:
+                lon_col = col
+                break
+        
+        # If lat/lon columns not found, try to extract from Point geometry
+        if (lat_col is None or lon_col is None) and all(geom.geom_type == 'Point' for geom in data.geometry):
+            data['lon'] = data.geometry.x
+            data['lat'] = data.geometry.y
+            lat_col = 'lat'
+            lon_col = 'lon'
+        
+        # If still no lat/lon, raise error
+        if lat_col is None or lon_col is None:
+            raise ValueError("Could not identify latitude and longitude columns.")
+        
+        # Standardize column names
+        data = data.rename(columns={lat_col: 'lat', lon_col: 'lon'})
+        
+        # Try to identify radius or diameter
+        radius_col = None
+        radius_candidates = ['radius', 'rad', 'r_km', 'r_m', 'radius_m', 'radius_km']
+        for col in radius_candidates:
+            if col in data.columns:
+                radius_col = col
+                break
+        
+        # If no radius, check for diameter
+        diameter_col = None
+        if radius_col is None:
+            diameter_candidates = ['diameter', 'diam', 'd_km', 'd_m', 'diameter_m', 'diameter_km']
+            for col in diameter_candidates:
+                if col in data.columns:
+                    diameter_col = col
+                    break
+        
+        # Process radius or diameter
+        if radius_col:
+            data = data.rename(columns={radius_col: 'radius'})
+        elif diameter_col:
+            data['radius'] = data[diameter_col] / 2
+            # Remove original diameter column to avoid confusing the constructor
+            data = data.drop(columns=[diameter_col])
+        else:
+            # If no radius or diameter, check if '_radius_m' exists (from previous export)
+            if '_radius_m' in data.columns:
+                pass  # Already has the right format
+            else:
+                raise ValueError("Could not identify radius or diameter column.")
+        
+        # Create and return a new CraterDatabase instance
+        return cls(data, body=body, units=units)
```

Comment:
ohhhh if we're passing to the constructor, all that column handling logic should already happen. and if it doesn't work for an imported geojson file, we should fix it where it fails in the constructor or in helpers.py, not with more custom logic here. all this method needs to do is try to figure out the body and units from the shapefile metadata or raise an error if missing. then pass the imported geodataframe to the constructor to handle over there